idunn/api/places_list.py

```diff
@@ -228,33 +190,61 @@ async def get_places_bbox_impl(
     )
 
 
-async def _fetch_places_list(params: PlacesQueryParam, poi_es_index_name: str):
-    if params.source == PoiSource.PAGESJAUNES:
-        return await run_in_threadpool(
-            pj_source.get_places_bbox,
-            params.category,
-            params.bbox,
-            size=params.size,
-            query=params.q,
-        )
-    if params.q:
-        # Default source (OSM) with query
-        bragi_response = await bragi_client.pois_query_in_bbox(
-            query=params.q, bbox=params.bbox, lang=params.lang, limit=params.size
-        )
-        return [BragiPOI(f) for f in bragi_response.get("features", [])]
+async def select_datasource(params):
+    if pj_source.bbox_is_covered(params.bbox):
+        await select_datasource_for_france(params)
+    else:
+        await select_datasource_outside_france(params)
 
-    # Default source (OSM) with category or class/subclass filters
-    if params.raw_filter:
-        filters = [MimirPoiFilter.from_url_raw_filter(f) for f in params.raw_filter]
+
+async def select_datasource_for_france(params):
+    if any(cat in params.category for cat in TRIPADVISOR_CATEGORIES_COVERED_IN_FRANCE):
+        params.source = PoiSource.TRIPADVISOR
+    elif await is_brand_detected_or_pj_category_cover(params):
+        params.source = PoiSource.PAGESJAUNES
     else:
-        filters = [f for c in params.category for f in c.raw_filters()]
-
-    bbox_places = await run_in_threadpool(
-        fetch_es_pois,
-        poi_es_index_name,
-        filters=filters,
-        bbox=params.bbox,
-        max_size=params.size,
-    )
-    return [POI(p["_source"]) for p in bbox_places]
+        params.source = PoiSource.OSM
+
+
+async def select_datasource_outside_france(params):
+    if any(cat in params.category for cat in TRIPADVISOR_CATEGORIES_COVERED_WORLDWIDE):
+        params.source = PoiSource.TRIPADVISOR
+    else:
+        params.source = PoiSource.OSM
+
+
+async def is_brand_detected_or_pj_category_cover(params):
+    return params.q or (params.category and all(c.pj_what() for c in params.category))
```

Comment:
This doesn't require to be async, which implies that `select_datasource_*` either